client/related_items/OutfitCard.jsx

```diff
@@ -1,14 +1,24 @@
 import React from 'react';
 
-const OutfitCard = (props) => (
-  <div className='relatedItemCard'>
-    <div id="action">Action</div>
-    <img id="thumbnail" src={props.productInfo.thumbnailUrl}></img>
-    <p id="category">{props.productInfo.category}</p>
-    <p id="name">{props.productInfo.name}</p>
-    <p id="price">{props.productInfo.price}</p>
-    <p id="rating">{props.productInfo.rating}</p>
-  </div>
-);
+class OutfitCard extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.removeFromOutfit = this.props.removeFromOutfit.bind(this);
+  }
+
+  render() {
+    return (
+      <div className='relatedItemCard' onClick={() => this.removeFromOutfit(this)}>
```

Comment:
Woah this took me a minute and I think I still might be shaky on it. Lemme see if I get it. You're passing down removeFromOutfit bound to the ItemsList so it's invoked in the context of that instance of the list and re-renders after the outfits are updated. But then you're binding each instance of an Outfit card to that previously bound function? If that's correct, I would imagine that on line 12 you don't could just do onClick=this.removeFromOutfit. I may be completely missing the point here, though. I trust that this is working as you expect but would be curious to chat about it with you just to clarify things for my own self!

Reply:
Ha yes this is a bit confusing. I didn't think it would work because I'm binding a function to two different components - what it's bound to depends on the context. So yes, I'm passing down a function that is bound to ItemsList (because it's changing the state of that component), but then I'm also binding it to each OutfitCard component because the only way I could figure out how to tell ItemsList which card to remove was by passing in the card that was clicked (the component, not the HTML element) as an argument to that function, so when we're back in ItemsList, we can use the id of that component (stored in its props) to know what to remove. 

Maybe there's a better way of doing this though - that was like my v1 approach and it worked! Happy to talk about this further today.